src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -55,12 +55,19 @@ internal class ModelCheckingStrategy(
     private lateinit var currentInterleaving: Interleaving
 
     override fun runImpl(): LincheckFailure? {
+        currentInterleaving = root.nextInterleaving() ?: return null
         while (usedInvocations < maxInvocations) {
-            // get new unexplored interleaving
-            currentInterleaving = root.nextInterleaving() ?: break
             usedInvocations++
             // run invocation and check its results
-            checkResult(runInvocation())?.let { return it }
+            val invocationResult = runInvocation()
+            if (suddenInvocationResult is SpinCycleFoundAndReplayRequired) {
```

Comment:
`is` or `===`?

Reply:
No matter as SpinCycleFoundAndReplayRequired is a Kotlin singleton object.